### PR TITLE
Metrics RT count optimization

### PR DESCRIPTION
**Pull request by mytang0**
## What is the purpose of the change
Optimize the logic of addRT and collectRT: 
1. Improve computing performance

old
```
    private final ConcurrentMap<M, AtomicLong>      lastRT = new ConcurrentHashMap<>();
    private final ConcurrentMap<M, LongAccumulator> minRT  = new ConcurrentHashMap<>();
    private final ConcurrentMap<M, AtomicLong>      totalRT = new ConcurrentHashMap<>();
    private final ConcurrentMap<M, AtomicLong>      rtCount = new ConcurrentHashMap<>();
```

new
```   
    // lastRT, totalRT, rtCount, avgRT share a container, can utilize the system cache line
    private final ConcurrentMap<M, AtomicLongArray> RTArray = new ConcurrentHashMap<>();
```
2. Optimize the interface design of MetricsCountSampler

## Checklist
- [ ] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [x] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by mytang0** [comment]
Please take a look   @songxiaosheng

**Comment by songxiaosheng** [comment]
> Please take a look   @songxiaosheng 

ok please wait

**Review by songxiaosheng — CHANGES_REQUESTED** [review]

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/SimpleMetricsCountSampler.java [hunk 1]
```diff
@@ -38,124 +43,120 @@
...
-    private final ConcurrentMap<M, AtomicLong>      totalRT = new ConcurrentHashMap<>();
-    private final ConcurrentMap<M, AtomicLong>      rtCount = new ConcurrentHashMap<>();
+    private final ConcurrentMap<M, AtomicLong> EMPTY_COUNT = new ConcurrentHashMap<>();
 
-    private Map<K, ConcurrentMap<M, AtomicLong>> metricCounter = new ConcurrentHashMap<>();
+    private final ConcurrentMap<M, LongAccumulator> minRT = new ConcurrentHashMap<>();
+    private final ConcurrentMap<M, LongAccumulator> maxRT = new ConcurrentHashMap<>();
+
+    // lastRT, totalRT, rtCount, avgRT share a container, can utilize the system cache line
+    private final ConcurrentMap<M, AtomicLongArray> RTArray = new ConcurrentHashMap<>();
```
[the remark this answers is not in the document]
Comment (mytang0) [reply dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/SimpleMetricsCountSampler.java]: Already fixed

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Pull request merged**